pkg/api/v1/deployment.go

```diff
@@ -132,6 +144,44 @@ func (g *DeploymentGroup) StopAllActiveDeployments(ctx echo.Context) error {
 	return g.stopDeployments(deployments, ctx)
 }
 
+func (g *DeploymentGroup) DeleteDeployment(ctx echo.Context) error {
+	cc, ok := ctx.(*auth.HttpAuthContext)
+	if !ok {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to get auth info")
+	}
+
+	deploymentId := ctx.Param("deploymentId")
+
+	// Get deployment
+	deploymentWithRelated, err := g.backendRepo.GetDeploymentByExternalId(ctx.Request().Context(), cc.AuthInfo.Workspace.Id, deploymentId)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Feels like we're leaking sql into the service. Can we return `nil, nil` when we don't find a deployment and there isn't an error? See my comment about the bad sql input.